File: nodejs-wdio/tests/specs/signInSuccess.spec.ts
import SignInPage from "../pages/signIn.page";
import WebinarsPage from "../pages/webinars.page";
import { expect } from "chai";
const fs = require("fs");

describe("Sign In Success", function () {
  before(function () {
    const file = fs.readFileSync("../fixtures/users.fixture.json");
    this.users = JSON.parse(file);
  });
  beforeEach(function () {
    SignInPage.visit();
    SignInPage.signIn(this.users["validUser"]);
  });
  afterEach(function () {
    // TO DO: add helper to logout with a http request
    WebinarsPage.signOut();
  });
  it("should be signed in", function () {
    expect(WebinarsPage.isSignedIn(), "User shoud be signed in").to.be.true;
  });
  it("should be redirect to Webinars page", function () {
    expect(
      () => WebinarsPage.waitForDisplayed(),
      "Webinars page should be displayed"
    ).not.to.throw();
  });
});

File: ruby-capybara/tests/specs/sign_in_success_spec.rb
require 'json'
require_relative '../pages/sign_in_page'
require_relative '../pages/webinars_page'
require_relative './spec_helper.rb'

describe 'Sign in Success', :shallow do
  before(:all) do
    file = File.read '../fixtures/users.fixture.json'
    @users = JSON.parse file
  end

  before(:each) do
    @sign_in = SignInPage.new
    @sign_in.visit
    @sign_in.sign_in(@users['validUser'])
    @webinars = WebinarsPage.new
  end

  after(:each) do
    @webinars.sign_out
  end

  it 'should be signed in' do
    expect(@sign_in.signed_in?)
      .to be_truthy
  end

  it 'should redirect to Webinars page' do
    expect { @webinars.wait_for_displayed }
      .not_to raise_error
  end
end
